examples/whl.py

```diff
@@ -0,0 +1,120 @@
+"""Install from a .whl file.
+"""
+
+import argparse
+import codecs
+import contextlib
+import sys
+import zipfile
+
+from installer._compat import pathlib
+from installer._compat.typing import TYPE_CHECKING
+from installer.layouts import DistInfo
+from installer.records import RecordItem, parse_record_file, write_record_file
+
+if TYPE_CHECKING:
+    from typing import Any, ContextManager, Dict, IO, Iterator
+
+
+class ZipFileInstaller(object):
+    """Install a local wheel.
+    """
+
+    def __init__(self, name, distinfo, zf):
+        # type: (str, DistInfo, zipfile.ZipFile) -> None
+        self._name = name
+        self._distinfo = distinfo
+        self._zf = zf
+
+    @classmethod
+    @contextlib.contextmanager
+    def create(cls, name, wheel_path):
+        # type: (str, pathlib.Path) -> Iterator[ZipFileInstaller]
+        project_name, project_version, _ = wheel_path.stem.split("-", 2)
+        with zipfile.ZipFile(str(wheel_path)) as zf:
+            entry_names = (name.lstrip("/").split("/", 1)[0] for name in zf.namelist())
+            distinfo = DistInfo.find(project_name, project_version, entry_names)
```

Comment:
nitpick: I'd prefer this to be ``dist_info``... even in the file name we split by ``-`` so don't see why we would not do the same with the variable name, makes it easier to read